packages/data/src/components/use-select/index.js

```diff
@@ -61,7 +62,16 @@ function Store( registry, suspense ) {
 			// in the interval between the `getValue` call during render and creating
 			// the subscription, which is slightly delayed. We need to ensure that this
 			// second `getValue` call will compute a fresh value.
-			lastMapResultValid = false;
+			lastMapResultValid =
+				activeStores.length === storeStatesOnMount.length &&
+				activeStores.every(
+					( storeName, index ) =>
+						storeStatesOnMount[ index ] ===
```

Comment:
The `subscribe` function could be called multiple times to obtain multiple subscriptions. React probably never does it, but it's a reasonable operation anyway. Here, the second `subscribe` call will dereference a `null` object and crash.

By the way, what if `storeStatesOnMount` was an object keyed by store name? Seems better than array to me.

Reply:
Yes, that's true, a Map is cleaner and also solves the other issue.